engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2673,7 +2683,12 @@ private LogicalOperatorInfo fillLogicalOperatorInfo(OperatorMeta operator)
     if (physicalOperators.size() > 0 && checkpointTimeAggregate.getAvg() != null) {
       loi.checkpointTimeMA = checkpointTimeAggregate.getAvg().longValue();
       loi.counters = latestLogicalCounters.get(operator.getName());
-      loi.autoMetrics = latestLogicalMetrics.get(operator.getName());
+      if (isPartialAggregates) {
+        loi.autoMetrics = latestLogicalMetrics.get(operator.getName());
```

Comment:
Why is return type different for complete and partial? window id should be available for both.

Reply:
Latest metrics might be partial or complete. If the latest metrics is partial and user querying the complete metrics, then the latest window id and the window id which belongs to complete metrics is different. That's why adding the window id to the complete query. Due to that different window id, user will be aware of that the latest metrics is not complete.

Reply:
How as a user I know that the latest is not complete? What if the latest is complete? I strongly prefer to unify format for complete and incomplete metrics and avoid using different return types.

Reply:
@chaithu14 Please reply or close the PR if you don't have time to look into it.